lxd/storage/drivers/generic_vfs.go

```diff
@@ -859,10 +879,8 @@ func genericVFSBackupUnpack(d Driver, sysOS *sys.OS, vol VolumeCopy, snapshots [
 			return nil, nil, err
 		}
 
-		snapVol, err := vol.NewSnapshot(snapName)
-		if err != nil {
-			return nil, nil, err
-		}
+		// Set the correct snapshots parent UUID.
+		snapVol.SetParentUUID(parentUUID)
```

Comment:
I think id prefer to see the snapVol initialisation done together in the loop above if thats ok?